corehq/apps/users/models.py

```diff
@@ -3133,3 +3139,33 @@ def deactivate_mobile_workers(cls, domain, date_deactivation):
         for chunked_ids in chunked(user_ids, 100):
             bulk_auto_deactivate_commcare_users(chunked_ids, domain)
             cls.objects.filter(domain=domain, user_id__in=chunked_ids).delete()
+
+    @classmethod
+    def update_trigger(cls, domain, user_id, deactivate_after):
+        existing_trigger = cls.objects.filter(domain=domain, user_id=user_id)
+        if isinstance(deactivate_after, str):
+            try:
+                parts = deactivate_after.split('-')
+                deactivate_after = date(int(parts[1]), int(parts[0]), 1)
```

Comment:
This code will happily parse formats other than `MM-YYYY`. For example, it would not complain about a date formatted as `MM-DD-YYYY` or `YYYY-MM-DD`, but the result would be wrong. Is this intended to do format validation?

A regular expression like `^(\d\d)-(\d{4})$` would give more confidence that the input has the expected format, and would produce easily parsed numeric outputs.

Might be worth moving this logic to a utility function, which could also be used in forms.py

https://github.com/dimagi/commcare-hq/blob/aa52af90aba7e6204d57d46d28fcf3872eb77cff/corehq/apps/users/forms.py#L100-L109